tests/testthat/test-split_functional_groups.R

```diff
@@ -0,0 +1,81 @@
+# Instructions ----
+#' This file follows the format generated by ecosystemdata:::use_testthat_template().
+#' Necessary tests include input and output (IO) correctness [IO
+#' correctness], edge-case handling [Edge handling], and built-in errors and
+#' warnings [Error handling]. See `?ecosystemdata:::use_testthat_template` for more
+#' information. Every test should have a @description tag that takes up just
+#' one line, which will be used in the bookdown report of {testthat} results.
+
+# Setup ----
+# Load or prepare any necessary data for testing
+
+# split_functional_groups ----
+## IO correctness ----
+test_that("split_functional_groups() works with correct inputs", {
+  #' @description Test that split_functional_groups(x) returns a tibble.
+  expect_equal(
+    object = split_functional_groups(c("spiny dogfish0", "spiny dogfish 2")),
+    expected = tibble::tibble(
+      functional_group = c("spiny dogfish0", "spiny dogfish 2"),
+      species = c("spiny dogfish", "spiny dogfish"),
+      group = c("0", "2")
+    )
+  )
+
+  #' @description Test that split_functional_groups(x) returns y.
+  expect_equal(
+    object = split_functional_groups(c(
+      "spiny dogfish 0",
+      "spiny dogfish 1-2",
+      "spiny dogfish +",
+      "apple"
+    )),
+    expected = tibble::tibble(
+      functional_group = c(
+        "spiny dogfish 0",
+        "spiny dogfish 1-2",
+        "spiny dogfish +",
+        "apple"
+      ),
+      species = c(
+        "spiny dogfish",
+        "spiny dogfish",
+        "spiny dogfish",
+        "apple"
+      ),
+      group = c("0", "1-2", "+", NA_character_)
+    )
+  )
+})
+
+## Edge handling ----
+test_that("split_functional_groups() returns correct outputs for edge cases", {
+  #' @description Test that split_functional_groups(x) uses sentence case rather
+  #' than camel case.
+  input <- c("SpinyDogfish0", "SpinyDogfish 2_3")
+  test <- split_functional_groups(input)
+  expect_equal(
+    object = test[["species"]],
+    expected = c("Spiny Dogfish", "Spiny Dogfish")
+  )
+  expect_equal(
```

Comment:
Could you add a one-sentence description for the test?

Reply:
done